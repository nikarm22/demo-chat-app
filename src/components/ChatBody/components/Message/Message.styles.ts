import { makeStyles, Theme } from "@material-ui/core";

export const useStyles = makeStyles((theme: Theme) => ({
    root: {
        display: 'block',
        marginTop: theme.spacing(1),
        maxWidth: '80%',
        alignSelf: 'flex-start',
        fontSize: '1.25rem',
        padding: theme.spacing(1),
        height: 'inherit',
        '& > span' : {
            whiteSpace: 'normal',
        },
    },
    left: {
        borderTopLeftRadius: 0,
    },
    right: {
        alignSelf: 'flex-end',
        borderBottomRightRadius: 0,
    }
}));
