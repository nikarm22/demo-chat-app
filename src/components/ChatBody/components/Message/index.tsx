import { Chip, makeStyles, Theme } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';

interface IMessage {
    side: 'left' | 'right';
    timestamp: string;
    children: string;
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        display: 'block',
        marginTop: theme.spacing(1),
        maxWidth: '80%',
        alignSelf: 'flex-start',
        fontSize: '1.25rem',
        padding: theme.spacing(1),
        height: 'inherit',
        '& > span' : {
            whiteSpace: 'normal',
        },
    },
    left: {
        borderTopLeftRadius: 0,
    },
    right: {
        alignSelf: 'flex-end',
        borderBottomRightRadius: 0,
    }
}));

const Message: React.FC<IMessage> = (props) => {
    const { side, timestamp, children } = props;
    const classes = useStyles();

    return <Chip
        className={clsx(
            classes.root,
            {
                [classes.right]: side === 'right',
                [classes.left]: side === 'left',
            },
        )}
        color={side === 'right' ? 'primary' : 'default'}
        label={children}
    />;
};

export { Message };
