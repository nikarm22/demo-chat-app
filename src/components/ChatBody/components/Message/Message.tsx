import Chip from '@material-ui/core/Chip';
import clsx from 'clsx';
import React from 'react';

import { useStyles } from './Message.styles';
interface IMessageProps {
    side: 'left' | 'right';
    timestamp: string;
    children: string;
};

const Message: React.FC<IMessageProps> = (props) => {
    const { side, children } = props;
    const classes = useStyles();

    return <Chip
        data-testid="message"
        className={clsx(
            classes.root,
            {
                [classes.right]: side === 'right',
                [classes.left]: side === 'left',
            },
        )}
        color={side === 'right' ? 'primary' : 'default'}
        label={children}
    />;
};

export { Message };
export type { IMessageProps };
