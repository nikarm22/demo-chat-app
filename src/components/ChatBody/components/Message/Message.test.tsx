import React from 'react';
import { render } from '@testing-library/react';
import { Message } from './Message';

describe('<Message />', () => {
    it('should render the message with correct text', async () => {
        const mockMessage = 'test';
        const { findByTestId } = render(
            <Message side='left' timestamp='00:00'>{mockMessage}</Message>
        );

        const element = await findByTestId('message');

        expect(element.textContent).toBe(mockMessage);
    });

    it('should render the message on correct side', async () => {
        const mockMessage = 'test';
        const { asFragment } = render(
            <Message side='right' timestamp='00:00'>{mockMessage}</Message>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
