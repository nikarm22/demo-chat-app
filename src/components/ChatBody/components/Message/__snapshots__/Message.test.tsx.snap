// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render the message on correct side 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root makeStyles-root-4 makeStyles-right-6 MuiChip-colorPrimary"
    data-testid="message"
  >
    <span
      class="MuiChip-label"
    >
      test
    </span>
  </div>
</DocumentFragment>
`;
